docs/how-to-use-monty/tutorials/implementing-actions.md

```diff
@@ -0,0 +1,161 @@
+
+# Sampling Actions
+
+For the **Motor System** to be able to sample the **Action** with the help of a sampler, you will need to include a `sample` method and declare an action-specific sampler protocol:
+
+```python
+class JumpActionSampler(Protocol):
+    def sample_jump(self, agent_id: str) -> Jump: ...
+
+class Jump(Action):
+
+    @staticmethod
+    def sample(agent_id: str, sampler: JumpActionSampler) -> Jump:
+        return sampler.sample_jump(agent_id)
+
+    def __init__(self, agent_id: str, how_high: float) -> None:
+        super().__init__(agent_id=agent_id)
+        self.how_high = how_high
+```
+
+This prepares the **Action** to be used in a sampler.
+
+The sampler itself will subclass `ActionSampler` and implement the requisite protocol:
+
+```python
+class MyConstantSampler(ActionSampler):
+
+    def __init__(
+        self,
+        actions: list[type[Action]] | None = None,
+        height: float = 1.8,
+        rng: Generator | None = None
+    ) -> None:
+        super().__init__(actions=actions, rng=rng)
+        self.height = 1.8
+
+    def sample_jump(self, agent_id: str) -> Jump:
+        return Jump(agent_id=agent_id, how_high=self.height)
+```
+
+The sampler then will be used along the lines of:
+
+```python
+sampler = MyConstantSampler(actions=[Jump])
+action = sampler.sample("agent_0")
+```
+
+For already existing samplers, see `src/tbp/monty/frameworks/actions/action_samplers.py`.
+
+# Actuating Actions
+
+For an **Action** to take effect within an **Environment**, it needs to be actuated. Include the `act` method and declare an action-specific actuator protocol:
+
+```python
+class JumpActuator(Protocol):
+    def actuate_jump(self, action: Jump) -> None: ...
+
+class Jump(Action):
+
+    def __init__(self, agent_id: str, how_high: float) -> None:
+        super().__init__(agent_id=agent_id)
+        self.how_high = how_high
+
+    def act(self, actuator: JumpActuator) -> None:
+        actuator.actuate_jump(self)
+```
+
+The actuator itself will be specific to the **Environment** and the simulator, robot, or otherwise. However, in general, it will implement the actuator protocol:
+
+```python
+class MyActuator:
+
+    def actuate_jump(self, action: Jump) -> None: ...
+        # custom code to make Jump happen
+```
+
+The **Environment**, simulator, or robot will invoke the actuator by calling the **Action**'s `act` method:
+
+```python
+SomeSimulatorAction = Union[Jump, Tumble] # Actions the simulator can actuate
+
+class SomeSimulator:
+
+    def step(self, action: SomeSimulatorAction) -> Observations
+        # ...
+        action.act(self.actuator)
+        # ...
+```
+
+You could implement the actuator as a Mixin, in that case it would look more like:
+
+```python
+class SomeSimulator(MyActuator):
+
+    def step(self, action: SomeSimulatorAction) -> Observations
+        # ...
+        action.act(self)
+        # ...
+```
+
+For an example of an actuator, see `src/tbp/monty/simulators/habitat/simulator.py`
```

Comment:
Nice tutorial! It helped me understand the new structure :) 
suggestion: One thing to consider could be making the example a bit more concrete. Like code that is concrete enough that someone could actually take it, know where to put it in Monty, and use it to add an action to one of our action spaces. For example, code that makes the camera in habitat move up by how_high amount and a sampler for that.